plumbing/object/commitgraph/commitnode_walker_date_order.go

```diff
@@ -0,0 +1,39 @@
+package commitgraph
+
+import (
+	"github.com/go-git/go-git/v5/plumbing"
+
+	"github.com/emirpasic/gods/trees/binaryheap"
+)
+
+// NewCommitNodeIterDateOrder returns a CommitNodeIter that walks the commit history,
+// starting at the given commit and visiting its parents in Committer Time and Generation order,
+// but with the  constraint that no parent is emitted before its children are emitted.
+//
+// This matches `git log --date-order`
+func NewCommitNodeIterDateOrder(c CommitNode,
+	seenExternal map[plumbing.Hash]bool,
```

Comment:
If all items in `seenExternal` are `true`, this could also be `map[plumbing.Hash]struct{}`.

Reply:
This was chosen to match the signature for NewCommitNodeIterCTime